parrot/src/pfs_table.cc

```diff
@@ -2056,7 +2107,7 @@ int pfs_table::locate( const char *n, char *buf, int length )
 		if(loc) delete(loc);
 		loc = 0;
 
-		if(resolve_name(0, n, &pname)) {
+		if(resolve_name(0, n, &pname, X_OK)) { //TODO check this
```

Comment:
Generally if there's an explicit mention in the documentation that no permissions are required on the item itself, I used F_OK (no checks) and something more restrictive on the parent. Also, for some of the Parrot-specific calls, I wasn't sure about the correct permissions, so I marked those parts with TODO and put something reasonable, or maybe a bit lenient